1-Assignment/chickenAndWolves.py

```diff
@@ -42,6 +42,49 @@ def isPuzzleEqual(x, goal):
         retval = False
     return retval
 
+#iterative deeping depth first search, takes in start state and end state
+def iddfs(start, goal):
+    closed = {}
+    startNode = Node(None, start)
+    fringe = deque([startNode])
+    counter = 0
+    dlimit = 1
+    while True:
+        while counter < dlimit:
+            if len(fringe) < 1:
+                return None
+            node = fringe.popleft()
+            if isPuzzleEqual(node.State, goal) is True:
+                print('expanded:', counter, "nodes")
+                return node
+            else:
+                counter += 1
+                if inClosed(closed, node.State) is False:
+                    addClosed(closed, node.State)
+                    fringe.extend(expandBfs(node))
+        dlimit = (dlimit * 2) + 1
+        fringe = deque([startNode])
+        closed = {}
+
+#depth first search, takes in start state and end state
+def dfs(start, goal):
+    closed = {}
+    startNode = Node(None, start)
+    fringe = deque([startNode])
+    counter = 0
+    while True:
+        if len(fringe) < 1:
+            return None
+        node = fringe.popleft()
+        if isPuzzleEqual(node.State, goal) is True:
+            print('expanded:', counter, "nodes")
+            return node
+        else:
+            counter += 1
+            if inClosed(closed, node.State) is False:
+                addClosed(closed, node.State)
+                fringe.extendleft(expandBfs(node))
```

Comment:
ok so this is a really weird bug but using popleft() and extendleft() causes an issue. Since your using those two methods the "back" of fringe is the first element. The successors array is built normally where the "back" is the back, so it ends up out of order